File: XiaoJuanSchoolPayment.Server/Data/DTO/StudentApplicationDTO.cs
using System.ComponentModel.DataAnnotations;

namespace XiaoJuanSchoolPayment.Server.Data.DTO
{
  public class StudentApplicationDTO
  {
    public Guid Id { get; set; }
    public required string StudentFirstName { get; set; }
    public required string StudentLastName { get; set; }
    public required string StudentName { get; set; }
    public string? StudentEmail { get; set; }
    public string? StudentPhone { get; set; }
    public bool HasStudentAccount { get; set; }
    public string EnrollmentType { get; set; } = "单人报名";
    public DateTime? EnrollmentDate { get; set; }
    public string VisaStatus { get; set; } = "未确认";
    public IList<StudentApplicationMemberDTO> Members { get; set; } = new List<StudentApplicationMemberDTO>();
    public IList<StudentApplicationPlanDTO> CoursePlans { get; set; } = new List<StudentApplicationPlanDTO>();
    public IList<StudentApplicationPlanDTO> AccommodationPlans { get; set; } = new List<StudentApplicationPlanDTO>();
    public int? StudyWeeks { get; set; }
    public Guid SchoolId { get; set; }
    public required string SchoolName { get; set; }
    public string? CourseName { get; set; }
    public string? AccommodationName { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public required string Status { get; set; }
    public string? StudentVisibleNotes { get; set; }
    public string? InternalNotes { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime LastUpdated { get; set; }
    public string ReviewStatus { get; set; } = "Published";
    public string? ChangeSummary { get; set; }
    public string? SubmittedByName { get; set; }
    public DateTime? SubmittedAt { get; set; }
    public string? PublishedByName { get; set; }
    public DateTime? PublishedAt { get; set; }
    public IList<StudentApplicationDocumentDTO> Documents { get; set; } = new List<StudentApplicationDocumentDTO>();
    public IList<StudentPaymentDTO> Payments { get; set; } = new List<StudentPaymentDTO>();
  }

  public class StudentApplicationMemberDTO
  {
    [Required, MaxLength(50)]
    public required string Key { get; set; }
    [Required, MaxLength(100)]
    public required string Name { get; set; }
    [MaxLength(50)]
    public string? Relationship { get; set; }
    [EmailAddress, MaxLength(256)]
    public string? Email { get; set; }
    [MaxLength(32)]
    public string? PhoneNumber { get; set; }
  }

  public class StudentApplicationPlanDTO
  {
    [Required, MaxLength(50)]
    public required string Key { get; set; }
    [Required, MaxLength(50)]
    public required string ParticipantKey { get; set; }
    [Required, MaxLength(200)]
    public required string Name { get; set; }
    public DateTime? StartDate { get; set; }
    [Range(1, 104)]
    public int? Weeks { get; set; }
    public DateTime? EndDate { get; set; }
  }

  public class StudentApplicationDocumentDTO
  {
    public Guid Id { get; set; }
    public required string DocumentType { get; set; }
    public required string DisplayName { get; set; }
    public required string OriginalFileName { get; set; }
    public long SizeBytes { get; set; }
    public bool IsVisibleToStudent { get; set; }
    public DateTime UploadedAt { get; set; }
    public required string DownloadUrl { get; set; }
  }

  public class CreateStudentApplicationDTO
  {
    [EmailAddress, MaxLength(256)]
    public string? Email { get; set; }
    [MaxLength(32)]
    public string? PhoneNumber { get; set; }
    [MaxLength(20)]
    public string EnrollmentType { get; set; } = "单人报名";
    public DateTime? EnrollmentDate { get; set; }
    [MaxLength(30)]
    public string VisaStatus { get; set; } = "未确认";
    public IList<StudentApplicationMemberDTO> Members { get; set; } = new List<StudentApplicationMemberDTO>();
    public IList<StudentApplicationPlanDTO> CoursePlans { get; set; } = new List<StudentApplicationPlanDTO>();
    public IList<StudentApplicationPlanDTO> AccommodationPlans { get; set; } = new List<StudentApplicationPlanDTO>();
    [Required, MaxLength(100)]
    public required string FirstName { get; set; }
    [MaxLength(100)]
    public string? LastName { get; set; }
    public Guid SchoolId { get; set; }
    [MaxLength(200)]
    public string? CourseName { get; set; }
    [MaxLength(200)]
    public string? AccommodationName { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    [MaxLength(50)]
    public string? Status { get; set; }
    [MaxLength(2000)]
    public string? StudentVisibleNotes { get; set; }
    [MaxLength(2000)]
    public string? InternalNotes { get; set; }
  }

  public class UpdateStudentApplicationDTO
  {
    [Required, MaxLength(100)]
    public required string FirstName { get; set; }
    [MaxLength(100)]
    public string? LastName { get; set; }
    [EmailAddress, MaxLength(256)]
    public string? Email { get; set; }
    [MaxLength(32)]
    public string? PhoneNumber { get; set; }
    [MaxLength(20)]
    public string EnrollmentType { get; set; } = "单人报名";
    public DateTime? EnrollmentDate { get; set; }
    [MaxLength(30)]
    public string VisaStatus { get; set; } = "未确认";
    public IList<StudentApplicationMemberDTO> Members { get; set; } = new List<StudentApplicationMemberDTO>();
    public IList<StudentApplicationPlanDTO> CoursePlans { get; set; } = new List<StudentApplicationPlanDTO>();
    public IList<StudentApplicationPlanDTO> AccommodationPlans { get; set; } = new List<StudentApplicationPlanDTO>();
    public Guid SchoolId { get; set; }
    [MaxLength(200)]
    public string? CourseName { get; set; }
    [MaxLength(200)]
    public string? AccommodationName { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    [Required, MaxLength(50)]
    public required string Status { get; set; }
    [MaxLength(2000)]
    public string? StudentVisibleNotes { get; set; }
    [MaxLength(2000)]
    public string? InternalNotes { get; set; }
  }

  public class StudentDocumentUploadDTO
  {
    public required IFormFile File { get; set; }
    [Required, MaxLength(50)]
    public required string DocumentType { get; set; }
    [MaxLength(255)]
    public string? DisplayName { get; set; }
    public bool IsVisibleToStudent { get; set; } = true;
  }

  public class StudentPaymentDTO
  {
    public Guid Id { get; set; }
    public Guid StudentApplicationId { get; set; }
    public required string StudentName { get; set; }
    public required string SchoolName { get; set; }
    public required string PayerName { get; set; }
    public required string PaymentType { get; set; }
    public decimal Amount { get; set; }
    public required string CurrencyCode { get; set; }
    public DateTime PaidAt { get; set; }
    public required string PaymentMethod { get; set; }
    public string? ReceivingAccount { get; set; }
    public string? ReferenceNumber { get; set; }
    public string? Note { get; set; }
    public required string ReviewStatus { get; set; }
    public string? SubmittedByName { get; set; }
    public DateTime SubmittedAt { get; set; }
    public string? ReviewedByName { get; set; }
    public DateTime? ReviewedAt { get; set; }
    public string? ReviewNote { get; set; }
    public required string OriginalFileName { get; set; }
    public long SizeBytes { get; set; }
    public bool IsVisibleToStudent { get; set; }
    public required string DownloadUrl { get; set; }
  }

  public class StudentPaymentSubmissionDTO
  {
    public IFormFile? File { get; set; }
    [Required, MaxLength(100)]
    public required string PayerName { get; set; }
    [Required, MaxLength(50)]
    public required string PaymentType { get; set; }
    [Range(typeof(decimal), "0.01", "9999999999999999")]
    public decimal Amount { get; set; }
    [Required, RegularExpression("^[A-Za-z]{3}$")]
    public required string CurrencyCode { get; set; }
    public DateTime PaidAt { get; set; }
    [Required, MaxLength(50)]
    public required string PaymentMethod { get; set; }
    [MaxLength(150)]
    public string? ReceivingAccount { get; set; }
    [MaxLength(100)]
    public string? ReferenceNumber { get; set; }
    [MaxLength(500)]
    public string? Note { get; set; }
  }

  public class StudentPaymentReviewDTO
  {
    [MaxLength(500)]
    public string? Note { get; set; }
  }

  public class ReturnStudentPaymentReviewDTO
  {
    [Required, MaxLength(500)]
    public required string Reason { get; set; }
  }

  public class SubmitStudentApplicationReviewDTO
  {
    [Required, MaxLength(500)]
    public required string ChangeSummary { get; set; }
  }

  public class PublishStudentApplicationDTO
  {
    [MaxLength(500)]
    public string? ChangeSummary { get; set; }
  }

  public class ReturnStudentApplicationReviewDTO
  {
    [Required, MaxLength(500)]
    public required string Reason { get; set; }
  }
}

File: xiaojuanschoolpayment.client/src/interfaces/student-application.dto.ts
export interface StudentApplicationDocumentDTO {
  id: string;
  documentType: string;
  displayName: string;
  originalFileName: string;
  sizeBytes: number;
  isVisibleToStudent: boolean;
  uploadedAt: string;
  downloadUrl: string;
}

export type StudentPaymentReviewStatus = 'PendingReview' | 'Confirmed' | 'Returned';

export interface StudentPaymentDTO {
  id: string;
  studentApplicationId: string;
  studentName: string;
  schoolName: string;
  payerName: string;
  paymentType: string;
  amount: number;
  currencyCode: string;
  paidAt: string;
  paymentMethod: string;
  receivingAccount?: string;
  referenceNumber?: string;
  note?: string;
  reviewStatus: StudentPaymentReviewStatus;
  submittedByName?: string;
  submittedAt: string;
  reviewedByName?: string;
  reviewedAt?: string;
  reviewNote?: string;
  originalFileName: string;
  sizeBytes: number;
  isVisibleToStudent: boolean;
  downloadUrl: string;
}

export interface StudentPaymentSubmissionDTO {
  payerName: string;
  paymentType: string;
  amount: number;
  currencyCode: string;
  paidAt: string;
  paymentMethod: string;
  receivingAccount?: string;
  referenceNumber?: string;
  note?: string;
}

export type StudentEnrollmentType = '单人报名' | '亲子家庭' | '多人同行';

export interface StudentApplicationMemberDTO {
  key: string;
  name: string;
  relationship?: string;
  email?: string;
  phoneNumber?: string;
}

export interface StudentApplicationPlanDTO {
  key: string;
  participantKey: string;
  name: string;
  startDate?: string;
  weeks?: number;
  endDate?: string;
}

export interface StudentApplicationDTO {
  id: string;
  studentFirstName: string;
  studentLastName: string;
  studentName: string;
  studentEmail?: string;
  studentPhone?: string;
  hasStudentAccount: boolean;
  enrollmentType: StudentEnrollmentType;
  enrollmentDate?: string;
  visaStatus: string;
  members: StudentApplicationMemberDTO[];
  coursePlans: StudentApplicationPlanDTO[];
  accommodationPlans: StudentApplicationPlanDTO[];
  studyWeeks?: number;
  schoolId: string;
  schoolName: string;
  courseName?: string;
  accommodationName?: string;
  startDate?: string;
  endDate?: string;
  status: string;
  studentVisibleNotes?: string;
  internalNotes?: string;
  createdAt: string;
  lastUpdated: string;
  reviewStatus: 'Draft' | 'PendingReview' | 'Published';
  changeSummary?: string;
  submittedByName?: string;
  submittedAt?: string;
  publishedByName?: string;
  publishedAt?: string;
  documents: StudentApplicationDocumentDTO[];
  payments: StudentPaymentDTO[];
}

export interface CreateStudentApplicationDTO {
  email?: string;
  phoneNumber?: string;
  enrollmentType: StudentEnrollmentType;
  enrollmentDate?: string;
  visaStatus: string;
  members: StudentApplicationMemberDTO[];
  coursePlans: StudentApplicationPlanDTO[];
  accommodationPlans: StudentApplicationPlanDTO[];
  firstName: string;
  lastName?: string;
  schoolId: string;
  courseName?: string;
  accommodationName?: string;
  startDate?: string;
  endDate?: string;
  status: string;
  studentVisibleNotes?: string;
  internalNotes?: string;
}

export interface UpdateStudentApplicationDTO {
  email?: string;
  phoneNumber?: string;
  enrollmentType: StudentEnrollmentType;
  enrollmentDate?: string;
  visaStatus: string;
  members: StudentApplicationMemberDTO[];
  coursePlans: StudentApplicationPlanDTO[];
  accommodationPlans: StudentApplicationPlanDTO[];
  firstName: string;
  lastName?: string;
  schoolId: string;
  courseName?: string;
  accommodationName?: string;
  startDate?: string;
  endDate?: string;
  status: string;
  studentVisibleNotes?: string;
  internalNotes?: string;
}

export const STUDENT_APPLICATION_STATUSES = [
  '资料准备',
  '已提交学校',
  '等待学校审核',
  '已收到录取通知书',
  '签证办理中',
  '行前准备',
  '已入学',
  '已完成',
  '保留/延期',
  '已取消',
] as const;

export const STUDENT_VISA_STATUSES = [
  '未确认',
  '无需办理',
  '未办理',
  '资料准备',
  '已递交',
  '审理中',
  '待补件',
  '已获签',
  '被拒/重新办理',
] as const;
